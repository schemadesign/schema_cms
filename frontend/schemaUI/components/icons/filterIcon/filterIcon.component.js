import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { getStyles } from './filterIcon.styles';
import FilterSVG from '../../../images/icons/filter.svg';
import { withStyles } from '../../styles/withStyles';

export class FilterIconComponent extends PureComponent {
  static propTypes = {
    customStyles: PropTypes.object,
  };

  render() {
    const { customStyles, theme, ...restProps } = this.props;
    const containerStyles = getStyles(theme);
    const styles = { ...containerStyles, ...customStyles };

    return <FilterSVG {...restProps} style={styles} />;
  }
}

export const FilterIcon = withStyles(FilterIconComponent);
