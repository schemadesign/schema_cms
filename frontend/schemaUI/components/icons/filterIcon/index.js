export { FilterIcon } from './filterIcon.component';
