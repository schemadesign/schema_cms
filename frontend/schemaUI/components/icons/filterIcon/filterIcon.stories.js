import React from 'react';
import { storiesOf } from '@storybook/react';

import { FilterIcon } from './filterIcon.component';
import { withTheme } from '../../../.storybook/decorators';

const defaultProps = { customStyles: { stroke: 'blue' } };

storiesOf('Icons/FilterIcon', module)
  .addDecorator(withTheme())
  .add('Default', () => <FilterIcon />)
  .add('with custom styles', () => <FilterIcon {...defaultProps} />);
