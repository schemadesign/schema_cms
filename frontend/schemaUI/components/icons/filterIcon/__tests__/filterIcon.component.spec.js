import React from 'react';
import { shallow } from 'enzyme';

import { FilterIconComponent } from '../filterIcon.component';

describe('FilterIcon: Component', () => {
  const defaultProps = {};

  const component = props => <FilterIconComponent {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom styles', () => {
    const customStyles = {
      fill: 'blue',
    };

    const wrapper = render({ customStyles });
    global.expect(wrapper).toMatchSnapshot();
  });
});
