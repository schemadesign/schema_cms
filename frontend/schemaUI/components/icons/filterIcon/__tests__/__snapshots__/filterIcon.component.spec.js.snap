// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterIcon: Component should render correctly 1`] = `
<filter.svg
  style={
    Object {
      "height": "60px",
      "stroke": "#FFF",
      "width": "60px",
    }
  }
/>
`;

exports[`FilterIcon: Component should render with custom styles 1`] = `
<filter.svg
  style={
    Object {
      "fill": "blue",
      "height": "60px",
      "stroke": "#FFF",
      "width": "60px",
    }
  }
/>
`;
